xcube/core/gen/gen.py

```diff
@@ -371,6 +373,9 @@ def step8(input_slice):
 def _update_cube_attrs(output_writer: DatasetIO, output_path: str,
                        global_attrs: Dict = None,
                        temporal_only: bool = False):
+    consolidated = os.path.exists(os.path.join(output_path, '.zmetadata'))
+    if consolidated:
+        optimize_dataset(input_path=output_path, in_place=True)
```

Comment:
```suggestion
        optimize_dataset(input_path=output_path, unchunk_coords=True, in_place=True)
```

...should avoid generation of cluttered coordinate variables. However, this may have a performance drawback. Could you please check?! Just looked into our `optimize_dataset` (actually `_unchunk_vars`); there is room for making more efficient.

Reply:
I created https://github.com/dcs4cop/xcube/pull/275 for this purpose.